torch/csrc/distributed/c10d/ProcessGroupNCCL.hpp

```diff
@@ -313,6 +313,16 @@ class TORCH_API ProcessGroupNCCL : public Backend {
     friend class ProcessGroupNCCL;
   };
 
+  class DebugInfoWriter {
```

Comment:
Move this class to `NCCLUtils.hpp` or even `Utils.hpp`? (depends on how general it is)

Reply:
I am not sure how much we benefit from moving it to `Utils.hpp` maybe `NCCLUtils.hpp`?

Reply:
Is there a reason this can't just be a `std::function<void(std::string&)>`?

Reply:
Actually, @wconstab and I had a conversation above. Originally I was using the function in the code but making a class can make it more generic and easier for internal to override?